tiledb/sm/misc/utils.h

```diff
@@ -292,6 +292,12 @@ void overlap(const T* a, const T* b, unsigned dim_num, T* o, bool* overlap);
 template <class T>
 double coverage(const T* a, const T* b, unsigned dim_num);
 
+/** Returns the intersection between r1 and r2. */
```

Comment:
Since this has a couple nested data structures in both the args and return type, can we expand the doc here?